lib/src/results.dart

```diff
@@ -207,3 +210,30 @@ extension RealmResultsInternal on RealmResults {
     return RealmResults<T>._(handle, realm);
   }
 }
+
+class RealmResultsIterator<T extends RealmObject> implements Iterator<T> {
+  final Iterable<T> _iterable;
+  final int _length;
+  int _index;
+
+  RealmResultsIterator(Iterable<T> iterable)
+      : _iterable = iterable,
+        _length = iterable.length,
+        _index = 0;
+
+  @override
+  T get current => (_index >= 0 && _index < _length) ? _iterable.elementAt(_index) : null as T;
```

Comment:
`null as T` will throw as T is not nullable. Maybe better to throw a `RangeError` in that case

Reply:
Shouldn't happen though.. so why have the check at all?

Reply:
Right, fixed in https://github.com/realm/realm-dart/pull/153/commits/86e68144180eff8d4aa10ca61940ee4dc50c4b8b